stdlib/test/builtin/test_simd.mojo

```diff
@@ -90,6 +90,18 @@ def test_convert_simd_to_string():
     )
 
 
+def test_simd_repr():
+    assert_equal(
```

Comment:
Can you please add tests for floating point types? And when you do, please ensure you cover special floating point values (nan, inf, -inf, -0.0).